### ISSUE TITLE
Add support for accessing content of positional / component objects ??

### ISSUE THREAD

**Issue by fingolfin** [oscar-system/GAP.jl#771]
We had need to access a member of a component object, to extract the collector of a pc group via  `G!.rewritingSystem` (I found no GAP function doing this). To my surprise, there is no GAP function for this, so I resorted to writing this:
```julia
_rewritingSystem = GAP.evalstr("x -> x!.rewritingSystem")
rewritingSystem(G::PcGroup) = _rewritingSystem(GAP.Globals.FamilyObj(one(G).X))
```
Perhaps this is the way to go; but perhaps there should be a "nice" way to access these things in GAP.jl? E.g. `getbangindex(G, sym)` for accessing `G!.sym` and similar `getbangproperty` and `set*` ?

**Comment by ThomasBreuer** [comment]
It is interesting that there is no official GAP function for this kind of access.
Apparently I have never needed this, and direct access to components is usually dangerous.
We could add
```
gap> BangComponent:= function( obj, nam ) return obj!.( nam ); end;;
gap> BangPosition:= function( obj, pos ) return obj![ pos ]; end;;
```
then the following works.
```
gap> BangComponent( SymmetricGroup( 4 ), "Size" );
24
gap> BangPosition( ZmodpZObj( 2, NextPrimeInt( 2^16 ) ), 1 );    
2
```
(No need to say that nobody promises that the `Size` value of a domain is stored in a component `Size`.)

**Comment by fingolfin** [comment]
Resolved by @ThomasBreuer via PR #816
